app/domain/util/concurrency_limited_cache.rb

```diff
@@ -66,5 +71,15 @@ def decrease_concurrent_requests
         )
       end
     end
+
+    def cached_key(args)
+      cache_key = args.key?(:cache_key) ? args.fetch(:cache_key) : args
```

Comment:
do we need this logic here? how about performing this check in the class that calls this class and send the correct cache key?

Reply:
other authenticators use it to and we didn't wanted to change singature for them